server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -114,12 +112,11 @@ public static boolean isFailureStoreFeatureFlagEnabled() {
     private final IndexMode indexMode;
     @Nullable
     private final DataStreamLifecycle lifecycle;
-    private final boolean rolloverOnWrite;
     private final boolean failureStoreEnabled;
-    private final List<Index> failureIndices;
     private volatile Set<String> failureStoreLookup;
```

Comment:
Should we move this lookup into `DataStreamIndices`? I feel like we could benefit from it on backing indices generally and it cleans up the class a bit more so that we don't have another view of the failure store indices hanging out in the main scope of the object.